providers/implementations/keymgmt/ml_dsa_kmgmt.c

```diff
@@ -258,52 +351,76 @@ static int ml_dsa_export(void *keydata, int selection,
 
     include_private = ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0);
 
-    /* Error if there is no public key */
-    if (ossl_ml_dsa_key_get_pub(key) == NULL)
-        return 0;
-
     /*
-     * Note that the private key always contains the public key elements so we
-     * just save the one blob and return.
+     * Note that the public key can be recovered from the private key, so we
+     * just export one or the other.  If the seed is present, both the seed and
```

Comment:
I dont know if this is the correct decision....

Reply:
I believe I retained the original behaviour (one blob or the other). We could export both, which is redundant for our providers, which always recomputed the public key, but arguably useful for exporting to others. I have no strong views on whether to include the public component in `export`. It is easier to add later, than remove, so starting with just one leaves more options open perhaps, otherwise toss a coin.

Reply:
I have raised an issue with a blocker on merge to master tag.

Reply:
not a blocker here..